### ISSUE TITLE
Programs-as-procedures

### ISSUE THREAD

**Issue by sorenbug** [PeterTillema/ICE#149]
I want to make a library for BigInts/BigRationals/BigDecimals, and I need to perform certain mathematical operations with them. I was thinking that a way to implement "subroutines" would be to allow programs themselves to accept arguments and then be able to call those programs from ICE, for example:
`SNBICMP`:
```ice
[i]INBICMP(BIGINTA,BIGINTB)

[i] some code here
SOMEMATH+SOMEOTHERMATH->RESULT
[i] some code here

RESULT
```
`SNBIADD`:
```ice
[i]INBIADD(BIGINTA,BIGINTB)

[i] some code here
prgmSNBICMP(MYBIGINTA,MYBIGINTB)->CMPRESULT
If CMPRESULT=-1
    SOMEMATH+SOMEOTHERMATH->RESULT
Else
    SOMEFOO+SOMEOTHERFOO->RESULT
End
[i] some code here

RESULT
```
I have another idea on how to handle parameters that I'll post in a separate issue.

**Comment by Overload02** [comment]
I strongly agree with this!! But it needs to be approached in a new way. 

Instead of having a separate program that allows you to run new command. it should be in source code of the program like in C.

This makes it much more easier and less space consuming. not only will this help, this will put an end to people asking for new commands in ICE. 😄  "Just A step closer to ICE 3.0!"

**Comment by sorenbug** [comment]
I agree that that's also a good idea, but (with my very limited knowledge of how ICE works) it seems like it would be harder to implement.

I think if it does end up being handled that way, having `import`s would be a great idea. I'd be able to fit a math library into a single program that others can just `import`.
